crates/rustc_codegen_spirv/src/builder/builder_methods.rs

```diff
@@ -1974,6 +1975,128 @@ impl<'a, 'tcx> BuilderMethods<'a, 'tcx> for Builder<'a, 'tcx> {
             // needing to materialize `&core::panic::Location` or `format_args!`.
             self.abort();
             self.undef(result_type)
+        } else if self.internal_buffer_load_id.borrow().contains(&llfn_def) {
```

Comment:
Clearly the code below should be factored out into a separate function - but for now this was easier during development.

Reply:
Done.